### PR TITLE
Mulit GPU + mask

### PR DESCRIPTION
**Pull request by philippgri11**

### CODE REVIEW

File: src/score_torch.py [hunk 1]
```diff
@@ -1,43 +1,39 @@
...
+        )
+        lower_right = torch.sqrt(
+            torch.mean((label_error - label_mean_error) ** 2, dim=(-1, -2))
+        )
+        return torch.mean(upper / (lower_left * lower_right))
+
+    def compute_weighted_mae(self, forecast, labels, lat_weights):
+        mask = ~torch.isnan(labels)
+        error = forecast - labels
+        mae = (torch.abs(error) * lat_weights * mask).mean()
```
Comment (max-ksoll) [inline src/score_torch.py]: Muss das mit der Mask nicht .sum()/mask.sum() sein?

File: src/score_torch.py [hunk 2]
```diff
@@ -1,43 +1,39 @@
...
+            (forecast_error - forecast_mean_error) * (label_error - label_mean_error),
+            dim=(-1, -2),
+        )
+        lower_left = torch.sqrt(
+            torch.mean((forecast_error - forecast_mean_error) ** 2, dim=(-1, -2))
+        )
+        lower_right = torch.sqrt(
+            torch.mean((label_error - label_mean_error) ** 2, dim=(-1, -2))
+        )
+        return torch.mean(upper / (lower_left * lower_right))
```
Comment (max-ksoll) [inline src/score_torch.py:43]: Hier auch wieder die Frage wie bei mae, muss hier nicht irgendwie die mask mit einbezogen werden in die mean berechnung?

File: src/score_torch.py [hunk 3]
```diff
@@ -1,43 +1,39 @@
...
-
-    lat_weights = torch.Tensor(np.cos(np.deg2rad(np.linspace(-90, 90, 121))))[:, None]
-
-    print(compute_weighted_acc(label, label, lat_weights, clim))
-    print(compute_weighted_acc(-label, label, lat_weights, clim))
-    print(compute_weighted_acc(forecast, label, lat_weights, clim))
+import numpy as np
+
+
+class WeightedMetrics:
```
Comment (max-ksoll) [inline src/score_torch.py]: Welchen Vorteil bietet die Klasse hier?

**Review by max-ksoll — CHANGES_REQUESTED** [review]
Bitte nächstes mal entweder vorher in einer PR oder danach in einer PR alle Files auto formatten

File: src/score_torch.py [hunk 4]
```diff
@@ -1,43 +1,50 @@
...
+
+    def compute_weighted_acc(self, forecast, labels, clim, lat_weights):
+        # input (bs, auto, 25, 121, 240)
+        # clim (25, 121, 240)
+        # lat_weights (121, 1)
+        mask = ~torch.isnan(labels)
+        forecast_error = (forecast - clim) * lat_weights
+        forecast_error[~mask] = 0  # Set forecast errors to 0 where mask is True
+        label_error = (labels - clim) * lat_weights
+        label_error = label_error.nan_to_num(nan=0)
```
Comment (max-ksoll) [inline src/score_torch.py]: warum zwei verschiedene Varianten um NaN auf 0 zu setzen?

File: src/score_torch.py [hunk 5]
```diff
@@ -1,43 +1,50 @@
...
+        assert torch.all(
+            (lower_left * lower_right) != 0
+        ), "lower_left * lower_right is zero, causing division by zero error."
+        return torch.mean(upper / (lower_left * lower_right))
+
+    def compute_weighted_mae(self, forecast, labels, lat_weights):
+        mask = ~torch.isnan(labels)
+        error = forecast - labels
+        mae = ((torch.abs(error) * lat_weights).nan_to_num(nan=0)).sum() / mask.sum(
+            dim=(-1, -2)
```
Comment (max-ksoll) [inline src/score_torch.py]: warum mask.sum nur über die letzten beiden Dimensionen?

File: src/score_torch.py (hunk 4 above)
Comment (philippgri11) [reply src/score_torch.py]: In den Labels steht nan drin aber im forecast warscheinlich nicht. Daher kann ich bei forecast nan nicht ersetzen.
